### PR TITLE
Show user location on map

### PR DESCRIPTION
**Pull request by i5glu**
#27 
I have implemented showing users location. Here a main scenarios:

1) if user is not on Texel Island, we set default region (as it did before, showing bungalows)
2) if user is on Texel Island and he is near bungalows, we set default region (as in scenario 1)
3) if user is on Texel Island, but further than default region, we calculate optimal region to show both user mark and bungalows

Also I used boolean flag to run setting region after receiving users location just for one time, so we don't update region after users location change (if needed, we can update region on every users location change). 
Text for Info.plist key is "We would like to show your location on map"

### CODE REVIEW

**Comment by funky-monkey** [comment]
Thanks for the pull-request! 👍 
Looks good - clear code style and very readable.
One small request though - can you also write tests to cover the above stated scenarios?

Looking forward to merging this PR!

Best, 
Sidney

File: SwiftIsland/Info.plist [hunk 1]
```diff
@@ -2,6 +2,8 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
+	<key>NSLocationWhenInUseUsageDescription</key>
+	<string>We would like to show your location on map</string>
```
Comment (ppeelen) [inline SwiftIsland/Info.plist]: Minor grammatical error (on *the* map), but also it maybe even better to change this to ```We would like to show you your location on Texel```?

**Comment by i5glu** [comment]
Not a problem, I will write some tests:)

**Review by ppeelen — CHANGES_REQUESTED** [review]
Great work! As @funky-monkey stated, some test would be awesome and a minor text change.

File: SwiftIsland/Info.plist (hunk 1 above)
Comment (ppeelen) [reply SwiftIsland/Info.plist]: @iglushchuk this comment is still left

Comment (i5glu) [reply SwiftIsland/Info.plist]: I did't forget about it. I am still working on test (want to cover scenarios on MapViewController). I will push tests with this text change.

Comment (ppeelen) [reply SwiftIsland/Info.plist]: Ah! Great :) Looking forward reading the test

Comment (i5glu) [reply SwiftIsland/Info.plist]: I pushed tests and text change) check it please when you have time

File: SwiftIsland/Info.plist [hunk 2]
```diff
@@ -2,6 +2,8 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
+	<key>NSLocationWhenInUseUsageDescription</key>
+	<string>We would like to show you your location on Texel</string>
```
Comment (SpacyRicochet) [inline SwiftIsland/Info.plist]: Maybe change this to explain slightly more?

> The Swift Island app uses your location to show you how to get to your sleeping accomodations on Texel.

Comment (i5glu) [reply SwiftIsland/Info.plist]: fixed

**Comment by i5glu** [comment]
Hi! I added some unit tests and change text in Info.plist, what about this pull request?

**Review by ppeelen — APPROVED** [review]
Nice work! Thanks a lot! 🌟🌟🌟

**Pull request merged**